backend/entityservice/database/models/models.py

```diff
@@ -13,6 +13,7 @@ class Encoding(Base):
 
     encoding_id = Column(BigInteger, primary_key=True)
     encoding = Column(LargeBinary, nullable=False)
+    dp = Column(ForeignKey('dataproviders.id', ondelete='CASCADE'))
```

Comment:
Being `nullable` was intentional so that users could upgrade?

Assuming we follow this up with a minor release, lets add a note that only "new" projects data gets cleaned up.